From 8a716a29721d900069d82878c45cf87a356adba3 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date: Tue, 17 Dec 2013 12:26:58 +0100
Subject: [PATCH] Fix QtWebKit build on ARM softfp

The low level interpreter depends on hardfp, so we need to disable it
when builing on ARM targets with softfp.

Task-number: QTBUG-35681
Change-Id: I8aaef546b92182a87cc7116ac452d8725de6a7db
Reviewed-by: Jocelyn Turcotte <jocelyn.turcotte@digia.com>
Signed-off-by: Fatih Aşıcı <fatih.asici@gmail.com>
---
 Source/WTF/wtf/Platform.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 3a9b509..cddc210 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -771,6 +771,11 @@
 #define ENABLE_LLINT 0
 #endif
 
+/* LLINT on ARM depends on an FPU */
+#if !defined(ENABLE_LLINT) && CPU(ARM) && !CPU(ARM_HARDFP)
+#define ENABLE_LLINT 0
+#endif
+
 /* On some of the platforms where we have a JIT, we want to also have the 
    low-level interpreter. */
 #if !defined(ENABLE_LLINT) \
-- 
1.8.5.1

